core/templates/components/button-directives/hint-and-solution-buttons.component.spec.ts

```diff
@@ -153,6 +156,18 @@ describe('HintAndSolutionButtonsComponent', () => {
       'content', audioTranslationLanguageService);
   });
 
+  it('should set .editorPreviewMode to the result', () => {
```

Comment:
@DhairyaMajmudar I don't really understand what this is supposed to be testing. Shouldn't we do something like:

- construct a StateCard **without** a solution, move forward a hint, and check that, when onNewCardOpened() is called, the hints reset (or that the first hint is the new card's first hint)
- construct a StateCard **with** a solution, move forward a hint, and check that, when onNewCardOpened() is called, the hints reset (or that the first hint is the new card's first hint)